src/ngraph/runtime/gpu/cudnn_emitter.cpp

```diff
@@ -1669,20 +1669,26 @@ size_t runtime::gpu::CUDNNEmitter::build_pooling(const cudnnPoolingMode_t& pool_
         {
             pool.reset(new gpu::primitive{
                 [=, &desc, &input_desc, &output_desc](void** inputs, void** outputs) {
+                    void* y = inputs[0];
+                    void* dy = inputs[0];
+                    void* x = inputs[0];
+                    if (pool_op == 0 || pool_op == 3)
+                    {
+                        y = inputs[2];
+                        dy = inputs[1];
+                        x = inputs[0];
+                    }
                     CUDNN_SAFE_CALL(cudnnPoolingBackward(*m_ctx->cudnn_handle,
```

Comment:
This checks to see if pooling is maxpool or avgpool. Eventually it would be better to separate build_pooling into separate functions for max and avg pooling. This replaces the previous call to the avg pool backprop kernel in which three copies of the input tensor were passed in: https://github.com/NervanaSystems/ngraph/pull/2115/files#diff-712d9cd1c68a53484d303bc6793ad1d2L324